p/scripts/main.js

```diff
@@ -169,16 +169,14 @@ function mark_favorite(active) {
 		return false;
 	}
 
-	var url = active.find("a.bookmark").attr("href");
```

Comment:
It looks like the previous version should work fine. No need to change anything in `main.js`, is there?

Reply:
That's what @Frenzie explained somewhere. It's better to rely on what we know (URL) and not the DOM content. If we do so, this part is not needed anymore.

Reply:
I'd say it was more of an opinion than an explanation but the gist is that I think the DOM is much more likely to break expectations for some reason than the URL. That applies not only to personal mods but also to the program itself.

Reply:
We can also include ISP modifying page content :cry:

Reply:
We were not relying on a random DOM structure (which might indeed change), but on a quite semantic class.

In this case, we end up having the URL in two places (in the HTML, and in the JavaScript), make assumptions about the URL format, and rely on the DOM even more (having an ID and a class on the active object) :-P

Reply:
Yes, a.bookmark should be fine. It's not child n of firstchild yada yada.

Reply:
So do I revert it?

Reply:
No objections either way in this case on my end. :-)

Reply:
Yes please, I would better like the previous version